Sprint 07

Dath-Remar opened the pull request.

Dath-Remar committed:
Покрыл Array+Extensions тестами

Dath-Remar committed:
Покрыл тестами MoviesLoader

Dath-Remar committed:
UITests написал + улучшил (заменил sleep)

Dath-Remar committed:
первый шаг рефакторинга «вынести метод convert() из MoveQuizViewController в Presenter» завершён

Dath-Remar committed:
Перенести логику yesButtonClicked(_ sender: UIButton) в presenter;
Также перенести логику noButtonClicked(_ sender: UIButton) в presenter;

Dath-Remar committed:
перенес в presenter     var currentQuestion: QuizQuestion?  и  var correctAnswers = 0

Dath-Remar committed:
перенес в presenter функцию didReceiveQuestion и отладил показ вопроса с изображением после загрузки приложения добавив Вызов completion callback

Dath-Remar committed:
перенес showAnswerResult

Dath-Remar committed:
перенес     var questionFactory: QuestionFactoryProtocol?

Dath-Remar committed:
presenter отформатировал

Dath-Remar committed:
сервис статистики перенес

Dath-Remar committed:
перенос handleDataLoadingError

Dath-Remar committed:
Добавление Reachability.swift + внесение в гитигнор

Dath-Remar committed:
init commit

Dath-Remar committed:
метод на основе ReachabilitySwift для проверки состояния сетевого соединения

Dath-Remar committed:
форматирование презентера

Dath-Remar committed:
MovieQuizViewController форматирование

Dath-Remar committed:
init commit

Dath-Remar committed:
Выполнил задачу по 7 спринту. Требует установки ПОДсов

andrew-kuzn-yndx commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,19 +1,19 @@
     // MARK: - IBOutlet
     
-    @IBOutlet private weak var noButton: UIButton!
-    @IBOutlet private weak var yesButton: UIButton!
```
а почему убрал private?)

Dath-Remar replied:
Хотел перенести контроль за нажатиями в Презенте, но это не его функция и забыл вернуть)))

andrew-kuzn-yndx commented on MovieQuiz/Services/QuestionFactoryDelegate.swift:
```diff
@@ -3,10 +3,10 @@ import Foundation
 
 protocol QuestionFactoryDelegate: AnyObject {
-    func didReceiveNextQuestion(question: QuizQuestion?)
+   // func didReceiveNextQuestion(question: QuizQuestion?)
```
Старайся не оставлять закомментированные строки)

Dath-Remar replied:
Понял, принял,исправил.

Dath-Remar replied:
Невероятная поддержка и мотивация!

Привет! @andrew-kuzn-yndx 

Огромная благодарность за твоё ревью моей проектной работы над MovieQuiz. В момент, когда я уже хотел всё бросить, твои слова поддержки и положительные комментарии дали мне силы и мотивацию продолжить.

Твоё внимание к деталям и конструктивная критика помогли мне улучшить мои навыки и повысили моральный дух. Я особенно ценю твои замечания по стилю кодирования и грамотное управление версиями, которые показывают твой высокий профессионализм.

Спасибо за то, что ты вдохновляешь и мотивируешь продолжать движение вперёд, твои ревью делают процесс обучения не только полезным, но и интересным. Жду новых вызовов и надеюсь на твою дальнейшую поддержку!

С уважением,
Сергей

andrew-kuzn-yndx commented:
Постфактум заметил один момент - после нажатия на ответ, кнопки Да / Нет блокироуются только до начала анимации
То есть когда начинается анимация смены постера, срабатывают нажатия на кнопки и цвет рамки можем меняться с зеленого на красный туда сюда
Посмотри этот момент плиз

Dath-Remar committed:
Исправление повторного нажатия кнопок во время переходов викторины
Путем корректировки потока управления для блокировки и последующей активации кнопок в подходящее время, мы гарантируем стабильность и ожидаемое поведение пользовательского интерфейса викторины при быстрых действиях пользователя.
+ вернул private и убрал закомментированный текст + форматирование кода

Dath-Remar commented:
> Постфактум заметил один момент - после нажатия на ответ, кнопки Да / Нет блокироуются только до начала анимации То есть когда начинается анимация смены постера, срабатывают нажатия на кнопки и цвет рамки можем меняться с зеленого на красный туда сюда Посмотри этот момент плиз

Поправил работу в потоках.Спасибо что подсветил это место. Ты самый классный ревьювер!!
